File: 프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 1;
    priority_queue pq(priorities.begin(), priorities.end());
    
    while(true) {
        if (pq.top() == priorities[0]) {
            if (location == 0)
                return answer;
            else {
                pq.pop();
                priorities.erase(priorities.begin());
                answer++;
            }
        }
        else if (pq.top() != priorities[0]) {
            if (location == 0)
                location = priorities.size();
            
            priorities.push_back(priorities[0]);
            priorities.erase(priorities.begin());
        }
        location--;
    }
    return answer;
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = 0
    while True:
        if priorities[0] == max(priorities):
            answer += 1
            if location == 0:
                return answer
            else:
                priorities.remove(priorities[0])
                location -= 1
        else:
            priorities.append(priorities[0])
            priorities.remove(priorities[0])
            if location == 0:
                location += len(priorities)-1
            else:
                location -= 1